### PR TITLE
Add Roku File System Panel and Roku App Overlays Panel

### PR DESCRIPTION
**Pull request by triwav**

### CODE REVIEW

File: src/viewProviders/RokuAppOverlaysViewViewProvider.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        this.registerCommandWithWebViewNotifier(context, VscodeCommand.rokuAppOverlaysViewRemoveAllOverlays);
+
+        subscriptions.push(vscode.commands.registerCommand(VscodeCommand.rokuAppOverlaysViewAddNewOverlay, async () => {
+            const options: vscode.OpenDialogOptions = {
+                canSelectMany: false,
+                openLabel: 'Add Overlay',
+                canSelectFiles: true,
+                canSelectFolders: false,
+                filters: {
+                    Images: ['png', 'jpg', 'jpeg', 'webp']
```
Comment (triwav) [inline src/viewProviders/RokuAppOverlaysViewViewProvider.ts:25]: Test with bmp and gif

File: src/viewProviders/RokuFileSystemViewViewProvider.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            const pathContentsInfo = message.context;
+            const result = await this.dependencies.rtaManager.onDeviceComponent.readFile({
+                path: pathContentsInfo.path
+            });
+
+            const filePath = path.join(os.tmpdir(), path.basename(pathContentsInfo.path));
+
+            // Write some content to the new file
+            fs.writeFileSync(filePath, result.binaryPayload);
+            await vscode.commands.executeCommand('vscode.open', vscode.Uri.file(filePath));
```
Comment (triwav) [inline src/viewProviders/RokuFileSystemViewViewProvider.ts:28]: Look into read only

Comment (triwav) [reply src/viewProviders/RokuFileSystemViewViewProvider.ts:28]: https://stackoverflow.com/a/76222660/1633757

File: webviews/src/shared/SortableGridHeader.svelte [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    export let sortDirection: 'none' | 'asc' | 'desc' = 'none';
+    export let column: number;
+
+    function onClick() {
+        dispatch('click');
+    }
+</script>
+
+<style>
+    vscode-data-grid-cell {
```
Comment (triwav) [inline webviews/src/shared/SortableGridHeader.svelte:17]: cursor: pointer

File: webviews/src/views/RokuFileSystemView/FileSystemEntry.svelte [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+<vscode-data-grid-row on:dblclick={onDoubleClick}>
+    <vscode-data-grid-cell grid-column="1">
+        {#if entry.type === 'file'}
+            <File />
+        {:else if entry.type === 'directory'}
+            <Folder />
+        {:else if entry.type === 'fileSystem'}
+            <Database />
+        {:else}
+            <Question />
```
Comment (triwav) [inline webviews/src/views/RokuFileSystemView/FileSystemEntry.svelte:55]: Test using loader

File: webviews/src/views/RokuFileSystemView/RokuFileSystemView.svelte [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+    function onOpen(event: CustomEvent<PathContentsInfo>) {
+        const pathContentsInfo = event.detail;
+        if (pathContentsInfo.type === 'file') {
+            intermediary.sendCommand(ViewProviderCommand.openRokuFile, pathContentsInfo);
+        } else {
+            updateCurrentPath(pathContentsInfo.path);
+        }
+    }
+
+    async function updateCurrentPath(path: string, force = false) {
```
Comment (triwav) [inline webviews/src/views/RokuFileSystemView/RokuFileSystemView.svelte:105]:
```suggestion
    async function updateCurrentPath(path: string, forceReload = false) {
```

Comment (triwav) [inline webviews/src/views/RokuFileSystemView/RokuFileSystemView.svelte:105]: Add documentation

File: webviews/src/views/RokuAppOverlaysView/RokuAppOverlaysView.svelte [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+            await odc.setValue({
+                keyPath: `${containerKeyPath}.#${overlay.id}.opacity`,
+                value: overlay.opacity
+            });
+        } else {
+            await conditionallyDeployOverlay(overlay, index);
+        }
+    }
+
+    async function deleteNode(id, parentKeypath = '') {
```
Comment (triwav) [inline webviews/src/views/RokuAppOverlaysView/RokuAppOverlaysView.svelte:154]: Add docs

**Review by TwitchBronBron — APPROVED** [review]
Looks great! awesome work.

**Pull request merged**